### DISCUSSION TITLE
Reg: User message when payload is empty

### DISCUSSION THREAD

**Discussion by ashoktronix** [phax/phase4#143] [Q&A]
Hi @phax ,
Do we have any reference or settings in phase4, When the payload is not part of attachment. Even though we set attachment to empty, Body is having extra elements starts from EncryptedData. 
IMHO these are generated by WSS4j , If you have any reference or settings to ignore EncryptedData in tag when the attachment is empty. 

```xml
	<S12:Body
		xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" wsu:Id="id-a5de2087-f7f4-4844-8e6e-09083811ddd2">
		<xenc:EncryptedData
			xmlns:xenc="http://www.w3.org/2001/04/xmlenc#" Id="ED-7e79a79a-1a9d-45a8-bf4d-bfe2ded08b2e" Type="http://www.w3.org/2001/04/xmlenc#Content">
			<xenc:EncryptionMethod Algorithm="http://www.w3.org/2009/xmlenc11#aes128-gcm"/>
			<ds:KeyInfo
				xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
				<wsse:SecurityTokenReference
					xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"
					xmlns:wsse11="http://docs.oasis-open.org/wss/oasis-wss-wssecurity-secext-1.1.xsd" wsse11:TokenType="http://docs.oasis-open.org/wss/oasis-wss-soap-message-security-1.1#EncryptedKey">
					<wsse:Reference URI="#EK-e5c82a31-681b-4bfe-b697-e4ef91a49b1b"/>
				</wsse:SecurityTokenReference>
			</ds:KeyInfo>
			<xenc:CipherData>
				<xenc:CipherValue>W18P87h/F9cz1M71HHQ8mfAe9Y7B8ojrRHYIIg==</xenc:CipherValue>
			</xenc:CipherData>
		</xenc:EncryptedData>
	</S12:Body>
``` 
But we are expecting like below:
```xml
	<S12:Body
		xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" wsu:Id="id-a5de2087-f7f4-4844-8e6e-09083811ddd2">
	</S12:Body>
```

**Comment by phax** [comment]
Well, if no attachments are present, and encryption is enabled, the Soap Body payload is encrypted.
In that case you need to disable encryption in your PMode.
Technically this can achieved when `AS4CryptParams.getAlgorithmCrypt()` returns `null`.
Not sure if that answers your question.

Can you please elaborate why you want to send an empty Soap Body without a Mime attachment? That seems to be an empty message....

**Reply by ashoktronix** [reply]
Requirement here is to send a sample test message which intern confirms about successful communication. With only signing. Kind of an HealthCheck.

**Reply by phax** [reply]
Okay, so something like the predefined test action?
![grafik](https://github.com/phax/phase4/assets/196219/e37744dc-e965-4496-a863-8abab915c1b2)

In that case I suggest you add a special PMode that disables encryption....

**Reply by ashoktronix** [reply]
We need to Sign User message and do not encrypt. So that the SOAP body will not have any security related entries. Can you give hint which PMode fullfills our requirement. To Sign only the user message and ignore the Encryption.

**Reply by phax** [reply]
Unfortunately, I don't have a ready made code snippet for this.
However, you have to make sure, that the `IPModeResolver` implementation finds it. Either via ID or from the combination of Service and Action. See `DefaultPModeResolver` for the default implementation

**Reply by ashoktronix** [reply]
Thanks for the hint.. We will work on that.. Its really helpful.
